tools/generate-logos

```diff
@@ -138,7 +138,8 @@ make_webp() {
 }
 
 make_one_android_icon() {
-    local src="$1" sourceset="$2" restype="$3" name="$4" size_px="$5" density="$6"
+    # SET BY CALLER: src sourceset restype name
```

Comment:
Interesting; `make_one_android_icon` uses these variables as they're defined in its caller, even when the caller uses `local`.

From the Bash Manual section 3.3 ([link](https://www.gnu.org/software/bash/manual/bash.html#Shell-Functions), formatting lost):

> The shell uses dynamic scoping to control a variable’s visibility within functions. With dynamic scoping, visible variables and their values are a result of the sequence of function calls that caused execution to reach the current function. The value of a variable that a function sees depends on its value within its caller, if any, whether that caller is the "global" scope or another shell function. This is also the value that a local variable declaration "shadows", and the value that is restored when the function returns.
> 
> For example, if a variable var is declared as local in function func1, and func1 calls another function func2, references to var made from within func2 will resolve to the local variable var from func1, shadowing any global variable named var.
> 
> The following script demonstrates this behavior. […]

Reply:
Yep. `local` protects the given variables between this function and its caller, but not between this function and its callees.

And actually it doesn't 100% protect the function from caring about what its caller might do with those variable names, either -- see this wrinkle in `tools/build-webview`:
```bash
readonly target dest check_path_name verbose

# …

sync() {
    # "_dest" because bash's `readonly` means can't shadow with `local`
    local src="$1" _dest="$2"
```

That is, even with `local` the function isn't totally free to pick a name of its choice without worrying about what names its calling code might have already used: if the caller has already used some name *and made it `readonly`*, then Bash won't let this function reuse the name even with `local`.

(This probably means it's not actually a good idea to routinely mark things `readonly`, as we do in that script.)